File: advanced_javascript/class/inheritance_super/inheritance_super.py
class Animal:
    def __init__(self, leg):
        self.leg = leg
    def printAnimal(self):
        print(self.name + "은 " + str(self.leg) + "개의 다리를 가진다")


class Lion(Animal):
    def __init__(self, name, leg):
        super().__init__(leg)   
        self.name = name 
        # self는 전달된 객체
        # self.name은 인스턴스 변수
    def getName(self):
        print("내 이름은 " + self.name)

myLion = Lion("King", 4)
myLion.getName()
myLion.printAnimal()

File: advanced_javascript/1.class/inheritance_super/inheritance_super.js
class Animal {
  constructor(leg) {
    this.leg = leg;
  }
  printAnimal() {
    console.log(`${this.name}은 ${this.leg}개의 다리를 가진다.`);
  }
}
class Lion extends Animal {
  constructor(name, leg) {
    super(leg);
    //부모 constructor의 input
    this.name = name;
    //생성자 함수 내의 this는 new 를 통해 만들어진 새로운 변수가 된다.
    //this.x -> 인스턴스 변수
  }
  getName() {
    console.log("내 이름은 " + this.name);
  }
}

myLion = new Lion("King", 4); //인스턴스 생성
myLion.getName();
myLion.printAnimal();
